enclave-runtime/src/initialization.rs

```diff
@@ -316,3 +331,49 @@ pub fn create_top_pool_author(
 		ocall_api,
 	))
 }
+
+/// Create a sel-signed certificate, signed with the enclave private key(Ed25519)
+pub fn enclave_self_signed_certificate() -> EnclaveResult<Certificate> {
+	let mut params: CertificateParams = Default::default();
+	params.not_before = date_time_ymd(2021, 05, 19);
+	params.not_after = date_time_ymd(4096, 01, 01);
+	let mut dn = DistinguishedName::new();
+	dn.push(DnType::OrganizationName, "Integritee Service");
+	dn.push(DnType::CommonName, DnValue::PrintableString("Enclave".to_string()));
+	params.distinguished_name = dn;
+
+	params.alg = &rcgen::PKCS_ED25519; //Signature Algorithm:
+
+	let private_key_der = generate_enclave_private_key_pkcs8_der()?;
+
+	let key_pair = rcgen::KeyPair::try_from(private_key_der.as_ref()).unwrap();
+	params.key_pair = Some(key_pair);
+
+	Certificate::from_params(params).map_err(|e| Error::Other(e.into()))
+}
+
+/// Save the private key in a PKCS#8 format. To be compatible with rcgen lib.
+///  PKCS#8 is specified in [RFC 5958].
+///
+/// [RFC 5958]: https://tools.ietf.org/html/rfc5958.
+pub fn generate_enclave_private_key_pkcs8_der() -> EnclaveResult<Vec<u8>> {
+	let signing = Ed25519Seal::unseal_from_static_file()?;
+	let private_key = signing.seed().as_slice();
+	let pk = signing.public().0;
+	let public_key = pk.as_slice();
+	let key_der = yasna::construct_der(|writer| {
+		writer.write_sequence(|writer| {
+			writer.next().write_u8(1);
+			// write OID
+			writer.next().write_sequence(|writer| {
+				writer.next().write_oid(&ObjectIdentifier::from_slice(ED25519));
+			});
+			let pk = yasna::construct_der(|writer| writer.write_bytes(&private_key));
+			writer.next().write_bytes(&pk);
+			writer.next().write_tagged(yasna::Tag::context(1), |writer| {
+				writer.write_bitvec(&BitVec::from_bytes(&public_key))
+			})
+		});
+	});
+	Ok(key_der)
+}
```

Comment:
Put these methods into the `tls_websocket_server` crate (new source file like `certificate_generation.rs`), not here in `initialization`. The signing key will then have to be a parameter that you pass into these methods. And also, don't make everything `pub` by default. Keep the scope as tight as possible (start with just `fn`, and then extend if needed, to `pub(crate)` or even `pub`)